httpserver/bin/hello_world_server.dart

```diff
@@ -7,13 +7,17 @@
 
 import 'dart:io';
 
-main() {
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4040)
-      .then((HttpServer server) {
-    print('listening on localhost, port ${server.port}');
-    server.listen((HttpRequest request) {
+main() async {
+  var requestServer =
+      await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4040);
+  print('listening on localhost, port ${requestServer.port}');
+
+  try {
+    await for (HttpRequest request in requestServer) {
       request.response.write('Hello, world!');
       request.response.close();
-    });
-  }).catchError((e) => print(e.toString()));
+    }
+  } catch (e) {
+    print(e.toString());
```

Comment:
But shouldn't we model best practice? I can add a "..." comment indicating that there might be more involved error processing.